hack/devtools/local_dev_env.sh

```diff
@@ -159,6 +158,13 @@ get_platform_workloadIdentity_role_sets() {
 assign_role_to_identity() {
     local objectId=$1
     local roleId=$2
+    
+    # if CLUSTER_RESOURCEGROUP is unset, it should probably be set to the cluster name
+    if [[ "${CLUSTER_RESOURCEGROUP}" == "" ]]; then
```

Comment:
Should we add an `if os.Getenv("CI")` to this if case as well? My thinking is that if it's CI e2e, we want cluster and resource group name to match. If not CI, we want the user to specify both.

Reply:
Thinking about this for a bit, i think the most straightforward way would be to just remove this check here and instead explicitly set `$CLUSTER_RESOURCEGROUP` to the cluster name in [run-rp-and-e2e.sh](https://github.com/Azure/ARO-RP/blob/db9f2241faf333ec890ecc19dc8fd57d2819fb56/hack/e2e/run-rp-and-e2e.sh#L15).

From a general design standpoint, i don't really like that it's not at all clear what setting `CI=true` does. So i'm more in favor of making those settings explicit.